packages/sitecore-jss/src/utils/timeout-promise.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * A helper to assign timeouts to fetch or other promises
+ * Useful in nextjs middleware until fetch.signal is fully supported by Vercel edge functions
+ */
+export default class TimeoutPromise {
+  timeoutID: NodeJS.Timeout | undefined;
+
+  constructor(private timeout: number | undefined) {
+    this.timeoutID = undefined;
+  }
+
+  /**
+   * Creates a timeout promise
+   */
+  get start(): Promise<any> {
+    if (this.timeout) {
```

Comment:
If we assume a non-undefined parameter, then can lose this if statement and the odd 'Timeout promise created with no timeout value' rejection.